modules/apps/app-builder/app-builder-web/src/main/java/com/liferay/app/builder/web/internal/deploy/ProductMenuAppDeployer.java

```diff
@@ -57,21 +62,46 @@ public void activate(BundleContext bundleContext) {
 	@Override
 	public void deploy(long appId) throws Exception {
 		String portletName = _getPortletName(appId);
-		String panelCategoryKey = _getPanelCategoryKey(appId);
+		String key = _getKey(appId);
 
 		AppBuilderApp appBuilderApp =
 			_appBuilderAppLocalService.getAppBuilderApp(appId);
 
 		String appName = appBuilderApp.getName(
 			LocaleThreadLocal.getDefaultLocale());
 
-		_serviceRegistrationsMap.computeIfAbsent(
-			appId,
-			key -> new ServiceRegistration<?>[] {
-				_deployAppPanelApp(portletName, panelCategoryKey),
-				_deployAppPanelCategory(appName, panelCategoryKey),
-				_deployAppPortlet(appName, portletName)
-			});
+		AppBuilderAppDeployment appBuilderAppDeployment =
+			_appBuilderAppDeploymentLocalService.getAppBuilderAppDeployment(
+				appId, "productMenu");
+
+		JSONObject jsonObject = _jsonFactory.createJSONObject(
+			appBuilderAppDeployment.getSettings());
+
+		JSONArray jsonArray = jsonObject.getJSONArray("deployments");
+
+		if (jsonArray.length() == 2) {
```

Comment:
there's a deployements array inside settings?

so, a AppDeployment has a Settings object which has another Deployments array inside it. we need to come up with a better name for that internal deployments array.

Reply:
This array is currently just for the Product Menu Deployer

Reply:
renamed to scope.